components/core/src/VariableDictionaryWriter.cpp

```diff
@@ -70,16 +69,17 @@ bool VariableDictionaryWriter::add_occurrence (const string& value, variable_dic
 
         // Insert the ID obtained from the database into the dictionary
         auto* entry = new VariableDictionaryEntry(value, id);
-        m_value_to_entry[value] = entry;
-        m_id_to_entry[id] = entry;
-
-        // Mark ID as dirty
-        m_uncommitted_entries.emplace_back(entry);
+        m_value_to_id[value] = id;
```

Comment:
Since the `entry`'s lifetime is within this block, we can use a local variable instead of a heap variable.